tsinfer/formats.py

```diff
@@ -1563,7 +1584,17 @@ def variants(self, inference_sites=None):
         for (j, genotypes), site in zip(
             self.genotypes(inference_sites), self.sites(inference_sites)
         ):
-            variant = Variant(site=site, alleles=site.alleles, genotypes=genotypes)
+            if site.time == self.USE_FREQ_AS_TIME:
```

Comment:
Maybe it's just me, but I find it clearer to assign the value for the nominal case like this:
```python
time = site.time
if site.time == self.USE_FREQ_AS_TIME:
    counts = allele_counts(genotypes)
    time = counts.derived / counts.known
```

It just seems a little clearer what's the nominal and exceptional cases here, and saves a little vertical space.

Reply:
Done